app/models/concerns/reviewable_action_builder.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ReviewableActionBuilder
+  extend ActiveSupport::Concern
+
+  # Build a single reviewable action and add it to the provided actions list.
+  # This is the canonical API used by both the legacy and refreshed UI code paths.
+  #
+  # @param actions [Reviewable::Actions] Actions instance to add to.
+  # @param id [Symbol] Symbol for the action, used to derive I18n keys.
+  # @param icon [String] Optional name of the icon to display with the action. Ignored in the refreshed UI.
+  # @param button_class [String] Optional CSS class for buttons in clients that render it.
+  # @param bundle [Reviewable::Actions::Bundle] Optional bundle object returned by add_bundle to group actions.
+  # @param client_action [String] Optional client-side action identifier (e.g. "edit").
+  # @param confirm [Boolean] When true, uses "reviewables.actions.<id>.confirm" for confirm_message.
```

Comment:
```suggestion
  # @param set_confirm_message [Boolean] When true, uses "reviewables.actions.<id>.confirm" for confirm_message.
```

Minor suggestion as I find `confirm` to be too generic and doesn't quite communicate the intent of the action to be taken when set.

Reply:
I'd intended it to be a drop-in replacement for the `build_action` method that a bunch of reviewables (both in core and plugins) define, which include this existing `confirm` parameter.

I don't mind renaming it, I don't have any strong feelings about whether I should mark the old param as deprecated, too.